app/assets/javascripts/Patient.js

```diff
@@ -0,0 +1,143 @@
+var Schema = mongoose.Schema;
+var Code = Schema.Types.Code;
+var Interval = Schema.Types.Interval;
+var Quantity = Schema.Types.Quantity;
+
+var PatientSchema = new Schema({
+  birth_datetime: DateTime,
+  qdm_version: { type: String, default: "5.3" },
+  ethnicity: Code,
+  race: Code,
+  sex: Code,
+  // These are the "data criteria", or QDM datatype elements that exist on a
+  // patient.
+  history_elements: []
+});
+
+// Returns an array of history elements that exist on this patient, that
+// match the given HQMF data criteria OID.
+PatientSchema.methods.get_by_hqmf_oid = function get_by_hqmf_oid(hqmf_oid, callback) {
+  return this.history_elements.filter(element => element.hqmf_oid === hqmf_oid);
+}
+
+// Returns an array of history elements that exist on this patient. Optionally
+// takes a category, which returns all history elements of that QDM category.
+// Example: patient.get_history_elements(category = 'encounters') will return
+// all Encounter QDM data types active on the patient.
+PatientSchema.methods.get_history_elements = function get_history_elements(category, callback) {
```

Comment:
would it make sense to enable querying by category and type (ordered/performed/etc.?)

so i could explicitly search for "medication performed" or "medication ordered".

as i'm writing this, i'm thinking yes, especially as these are different models and not being able to do a direct query for them would mean I'd have to do some additional processing to figure out exactly what I'm looking at.

Reply:
Makes sense, thinking about the above comment (adding "types").